File: mvvm/src/main/java/com/base/mvvm/feature/login/LoginHandler.java
package com.base.mvvm.feature.login;

import android.view.View;

import com.base.mvvm.databinding.ActivityLoginBinding;

/**
 * 描述: 点击事件
 *
 * @author : <a href="mailto:fanhuayong@yinli56.com">Fanhy</a>
 * @version : Ver 1.0
 * @date : 2018-04-18 10:07
 */

public class LoginHandler {
    private ActivityLoginBinding binding;
    private LoginViewModel viewModel;

    public LoginHandler(ActivityLoginBinding binding) {
        this.binding = binding;
        viewModel = new LoginViewModel(binding);
    }

    public void loginDo(View view) {
        viewModel.login(binding.getUserBean());
    }
}

File: kotlindemo/src/main/java/com/base/demo/login/LoginHandler.kt
package com.base.demo.login

import android.view.View

/**
 * 描述: 登陆处理
 *
 * @author : <a href="mailto:fanhuayong@yinli56.com">Fanhy</a>
 * @version : Ver 1.0
 * @date : 2018-05-11 11:29
 */
class LoginHandler {
    var act: LoginActivity ?= null

    constructor(act: LoginActivity) {
        this.act = act
    }

    fun loginClick(view: View) {
        var userBean = act?.loginBinding?.userBean
        act?.vm?.login(userBean?.userName as String, userBean.userPwd)
    }
}
